src/Rectangle.php

```diff
@@ -0,0 +1,21 @@
+<?php
+namespace Shape;
+
+class Rectangle extends Square implements ShapeInterface
```

Comment:
Nice! Both extends & implements, passes length to parent and keeps width in this class and uses both for area calculation.

One question to find an answer to - if parent class (Square) already implements some interface - would it mean that any class (in your example Rectangle) automatically implements it?